model/task/generate_tasks_estimations.go

```diff
@@ -1,55 +1,101 @@
 package task
 
 import (
+	"fmt"
+	"math"
+	"time"
+
 	"github.com/evergreen-ci/evergreen"
-	"github.com/evergreen-ci/evergreen/db"
 	"github.com/evergreen-ci/utility"
-	adb "github.com/mongodb/anser/db"
 	"github.com/pkg/errors"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
 const (
-	numRevisionsToSearch = 10
+	lookBackTime = 7 * 24 * time.Hour // one week
 )
 
+type generateTasksEstimationsResults struct {
+	DisplayName        string  `bson:"_id"`
+	EstimatedCreated   float64 `bson:"est_created"`
+	EstimatedActivated float64 `bson:"est_activated"`
+}
+
 func (t *Task) setGenerateTasksEstimations() error {
-	if !t.GenerateTask || (t.EstimatedNumGeneratedTasks != nil && t.EstimatedNumActivatedGeneratedTasks != nil) {
+	// Do not run if estimations have already been cached.
+	if t.EstimatedNumGeneratedTasks != nil && t.EstimatedNumActivatedGeneratedTasks != nil {
 		return nil
 	}
 
-	query := db.Query(
-		ByPreviousCommit(
-			t.BuildVariant,
-			t.DisplayName,
-			t.Project,
-			evergreen.RepotrackerVersionRequester,
-			t.RevisionOrderNumber,
-		),
-	).Sort([]string{"-" + RevisionOrderNumberKey}).Limit(numRevisionsToSearch)
+	match := bson.M{
```

Comment:
nit: might be a little better for code organization to move the long aggregation into a helper function in `model/task/db.go` and then just return the `[]generateTasksEstimationResults`.

Reply:
makes sense, moved over
(lookBack is still in this file because it's a var only used for seeing the estimation. Would it be better practice to move it to db?)

Reply:
That seems fine since, even if there's only one usage, the lookback constant is specific to this function.